src/core/MIGRATION.md

```diff
@@ -1243,6 +1244,43 @@ class MyPlugin {
   }
 ```
 
+If your plugin also have a client-side part, you can also expose configuration properties to it using a whitelisting mechanism with the configuration `exposeToBrowser` property.
+```typescript
+// my_plugin/server/index.ts
+import { schema, TypeOf } from '@kbn/config-schema';
+import { PluginConfigDescriptor } from 'kibana/server';
+
+const configSchema = schema.object({
+  secret: schema.string({ defaultValue: 'Only on server' }),
+  uiProp: schema.string({ defaultValue: 'Accessible from client' }),
+});
+
+type ConfigType = TypeOf<typeof configSchema>;
+
+export const config: PluginConfigDescriptor<ConfigType> = {
+  exposeToBrowser: ['uiProp'],
+  schema: configSchema,
+};
+
+export type ClientConfigType = Pick<ConfigType, 'uiProp'>
```

Comment:
I'd avoid providing an example exporting types from `server/index.ts`. This is a plugin entry point. It means that plugin code with all nodejs dependencies will be bundled in the client bundle.

Reply:
You are right, this is a terrible thing to put in the migration guide.